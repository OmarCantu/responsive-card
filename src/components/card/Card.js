import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import * as colors from '../../constants/colors';
import { LOGO_PROPTYPE, TILE_PROPTYPE } from '../../constants/propTypes';

const Wrapper = styled.div`
  display: inline-block;
  max-width: 600px;
  width: ${props => props.width};
`;

const Container = styled.div`
  overflow: hidden;
  position: relative;
`;

const Tile = styled.img.attrs({
  role: 'presentation'
})`
  display: block;
  max-width: 100%;
  transition: transform .5s ease;

  :hover {
    transform: scale(1.1);
  }
`;

const Overlay = styled.div`
  align-items: center;
  background-color: ${colors.BLACK50};
  bottom: 0;
  display: grid;
  grid-template-rows: 100%;
  grid-template-columns: 13% 87%;
  height: 31%;
  padding: 2.5%;
  position: absolute;
  width: 100%;
`;

const Logo = styled.img.attrs({
  role: 'presentation'
})`
  height: auto;
  max-height: 100%;
  max-width: 100%;
`;

const Title = styled.h2`
  color: ${colors.WHITE};
  overflow: hidden;
  padding-left: 3.5%;
  white-space: nowrap;

  @media (max-width: 479.5px) {
    font-size: 1.3rem;
  }

  @media (min-width: 480px) and (max-width: 767.5px) {
    font-size: 1.6rem;
  }

  @media (min-width: 768px) {
    font-size: 2.2rem;
  }
`;

const Card = props => {
  const { 
    className, 
    logo: { alt: logoAlt, src: logoSrc },
    tile: { alt: tileAlt, src: tileSrc }, 
    title,
    width
  } = props;

  return (
    <Wrapper className={className} width={width}>
      <Container>
        <Tile 
          alt={tileAlt}
          src={tileSrc} 
          title={tileAlt}
        />

        {title && (
          <Overlay>
            <Logo 
              alt={logoAlt} 
              src={logoSrc} 
              title={logoAlt}
            />

            <Title>
              {title}
            </Title>
          </Overlay>
        )}
      </Container>
    </Wrapper>
  )
};

Card.propTypes = {
  className: PropTypes.string,
  logo: LOGO_PROPTYPE,
  tile: TILE_PROPTYPE.isRequired,
  title: PropTypes.string,
  width: PropTypes.string,
};

Card.defaultProps = {
  className: undefined,
  logo: LOGO_PROPTYPE.default,
  title: undefined,
  width: 'auto',
};

export default Card;