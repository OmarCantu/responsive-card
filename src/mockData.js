import tileSrc from './static/images/tile.jpg';

export const CARD_DATA = {
  tile: {
    alt: 'Home and Away',
    src: tileSrc
  },
  title: 'Home and Away'
}