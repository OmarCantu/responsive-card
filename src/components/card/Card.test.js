import React from 'react';
import { mount } from 'enzyme';

import Card from './Card';
import { CARD_DATA } from '../../mockData';

const { tile, title } = CARD_DATA;
const wrapper = mount(<Card className='card' tile={tile} />);

test('Card has class "card"', () => {
  expect(wrapper.find('Card[className="card"]')).toHaveLength(1);
});

test('Card has only one img tag without a title prop defined', () => {
  expect(wrapper.find('img')).toHaveLength(1);
});

test('Card has no h2 tag without a title prop defined', () => {
  expect(wrapper.find('h2')).toHaveLength(0);
});

test('Card has two img tags with a title prop defined', () => {
  wrapper.setProps({ title });
  expect(wrapper.find('img')).toHaveLength(2);
});

test('Card has one h2 tag with a title prop defined', () => {
  expect(wrapper.find('h2')).toHaveLength(1);
});

test('Component matches Snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});