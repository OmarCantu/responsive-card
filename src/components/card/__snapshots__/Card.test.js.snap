// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches Snapshot 1`] = `
<Card
  className="card"
  logo={
    Object {
      "alt": "Seven logo",
      "src": "logo.png",
    }
  }
  tile={
    Object {
      "alt": "Home and Away",
      "src": "tile.jpg",
    }
  }
  title="Home and Away"
  width="auto"
>
  <styled.div
    className="card"
    width="auto"
  >
    <StyledComponent
      className="card"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "ipPoCj",
            "rules": Array [
              "
  display: inline-block;
  max-width: 600px;
  width: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      width="auto"
    >
      <div
        className="card sc-bdVaJa ipPoCj"
        width="auto"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "geMaRF",
                  "rules": Array [
                    "
  overflow: hidden;
  position: relative;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH geMaRF"
            >
              <styled.img
                alt="Home and Away"
                src="tile.jpg"
                title="Home and Away"
              >
                <StyledComponent
                  alt="Home and Away"
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [
                        Object {
                          "role": "presentation",
                        },
                      ],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": true,
                        "lastClassName": "gUYLkx",
                        "rules": Array [
                          "
  display: block;
  max-width: 100%;
  transition: transform .5s ease;

  :hover {
    transform: scale(1.5);
  }
",
                        ],
                      },
                      "displayName": "styled.img",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": "img",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  src="tile.jpg"
                  title="Home and Away"
                >
                  <img
                    alt="Home and Away"
                    className="sc-htpNat gUYLkx"
                    role="presentation"
                    src="tile.jpg"
                    title="Home and Away"
                  />
                </StyledComponent>
              </styled.img>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": true,
                        "lastClassName": "eBpJLS",
                        "rules": Array [
                          "
  align-items: center;
  background-color: ",
                          "rgba(0, 0, 0, .5)",
                          ";
  bottom: 0;
  display: grid;
  grid-template-rows: 100%;
  grid-template-columns: 13% 87%;
  height: 31%;
  padding: 2.5%;
  position: absolute;
  width: 100%;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-bxivhb eBpJLS"
                  >
                    <styled.img
                      alt="Seven logo"
                      src="logo.png"
                      title="Seven logo"
                    >
                      <StyledComponent
                        alt="Seven logo"
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [
                              Object {
                                "role": "presentation",
                              },
                            ],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-ifAKCX",
                              "isStatic": true,
                              "lastClassName": "VNtXC",
                              "rules": Array [
                                "
  height: auto;
  max-height: 100%;
  max-width: 100%;
",
                              ],
                            },
                            "displayName": "styled.img",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-ifAKCX",
                            "target": "img",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        src="logo.png"
                        title="Seven logo"
                      >
                        <img
                          alt="Seven logo"
                          className="sc-ifAKCX VNtXC"
                          role="presentation"
                          src="logo.png"
                          title="Seven logo"
                        />
                      </StyledComponent>
                    </styled.img>
                    <styled.h2>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-EHOje",
                              "isStatic": true,
                              "lastClassName": "gNASJJ",
                              "rules": Array [
                                "
  color: ",
                                "rgb(255, 255, 255)",
                                ";
  overflow: hidden;
  padding-left: 3.5%;
  white-space: nowrap;

  @media (max-width: 479.5px) {
    font-size: 1.3rem;
  }

  @media (min-width: 480px) and (max-width: 767.5px) {
    font-size: 1.6rem;
  }

  @media (min-width: 768px) {
    font-size: 2.2rem;
  }
",
                              ],
                            },
                            "displayName": "styled.h2",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-EHOje",
                            "target": "h2",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <h2
                          className="sc-EHOje gNASJJ"
                        >
                          Home and Away
                        </h2>
                      </StyledComponent>
                    </styled.h2>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Card>
`;
